libs/wingc/src/jsify/snapshots/qualify_inflight_type_refrencing_preflight_instance.snap

```diff
@@ -0,0 +1,209 @@
+      constructor($scope, $id, ) {
+        super($scope, $id);
+      }
+      static _toInflightType() {
+        return `
+          require("${$helpers.normalPath(__dirname)}/inflight.PreflightC-1.cjs")({
+          })
+        `;
+      }
+      _toInflight() {
+        return `
+          (await (async () => {
+            const PreflightCClient = ${PreflightC._toInflightType()};
+            const client = new PreflightCClient({
+            });
+            if (client.$inflight_init) { await client.$inflight_init(); }
+            return client;
+          })())
+        `;
+      }
+      get _liftMap() {
+        return ({
+          "bar": [
+          ],
+          "$inflight_init": [
+          ],
+        });
+      }
+    }
+    class InflightC extends $stdlib.std.Resource {
+      constructor($scope, $id, ) {
+        super($scope, $id);
+      }
+      static _toInflightType() {
+        return `
+          require("${$helpers.normalPath(__dirname)}/inflight.InflightC-1.cjs")({
+            $pc: ${$stdlib.core.liftObject(pc)},
+          })
+        `;
+      }
+      _toInflight() {
+        return `
+          (await (async () => {
+            const InflightCClient = ${InflightC._toInflightType()};
+            const client = new InflightCClient({
+            });
+            if (client.$inflight_init) { await client.$inflight_init(); }
+            return client;
+          })())
+        `;
+      }
+      get _liftMap() {
+        return ({
+          "foo": [
+            [pc, ["bar"]],
+          ],
+          "$inflight_init": [
+            [pc, []],
+          ],
+        });
+      }
+    }
+    if ($preflightTypesMap[143]) { throw new Error("InflightC is already in type map"); }
+    $preflightTypesMap[143] = InflightC;
+    class $Closure1 extends $stdlib.std.AutoIdResource {
+      _id = $stdlib.core.closureId();
+      constructor($scope, $id, ) {
+        super($scope, $id);
+        $helpers.nodeof(this).hidden = true;
+      }
+      static _toInflightType() {
+        return `
+          require("${$helpers.normalPath(__dirname)}/inflight.$Closure1-1.cjs")({
+            $InflightC: ${$stdlib.core.liftObject(InflightC)},
+          })
+        `;
+      }
+      _toInflight() {
+        return `
+          (await (async () => {
+            const $Closure1Client = ${$Closure1._toInflightType()};
+            const client = new $Closure1Client({
+            });
+            if (client.$inflight_init) { await client.$inflight_init(); }
+            return client;
+          })())
+        `;
+      }
+      get _liftMap() {
+        return ({
+          "handle": [
+            [$helpers.nodeof(this).root.$preflightTypesMap[143]._singleton(this,"InflightC_singleton_143"), ["foo"]],
+            [InflightC, []],
+          ],
+          "$inflight_init": [
+            [$helpers.nodeof(this).root.$preflightTypesMap[143]._singleton(this,"InflightC_singleton_143"), []],
```

Comment:
It might be possible to shorten this to something like
```suggestion
            [$helpers.inflightSingleton(this,"InflightC_singleton", 143), []],
```

Reply:
cleaned this up a bit